src/components/measurements/index.js

```diff
@@ -0,0 +1,279 @@
+      for (const [field, values] of Object.entries(activeFilters)) {
+        if (values.length > 0 && !values.includes(measurement[field])) return false;
+      }
+      return true;
+    })
+  };
+};
+
+const MeasurementsPlot = ({height, width, showLegend, setPanelTitle}) => {
+  // Use `lodash.isEqual` to deep compare object states to prevent unnecessary re-renderings of the component
+  const { treeStrainVisibility, treeStrainColors } = useSelector((state) => treeStrainPropertySelector(state), isEqual);
+  const groupBy = useSelector((state) => state.controls.measurementsGroupBy);
+  const filters = useSelector((state) => state.controls.measurementsFilters);
+  const display = useSelector((state) => state.controls.measurementsDisplay);
+  const showOverallMean = useSelector((state) => state.controls.measurementsShowOverallMean);
+  const showThreshold = useSelector((state) => state.controls.measurementsShowThreshold);
+  const collection = useSelector((state) => state.measurements.collectionToDisplay, isEqual);
+  const { title, x_axis_label, threshold, fields, measurements, groupings } = collection;
+
+  // Ref to access the D3 SVG
+  const d3Ref = useRef(null);
+
+  // State for storing data for the HoverPanel
+  const [hoverData, setHoverData] = useState(null);
+
+  // Filter and group measurements
+  const {activeFilters, filteredMeasurements} = filterMeasurements(measurements, treeStrainVisibility, filters);
+  const groupingOrderedValues = groupings.get(groupBy).values;
+  const groupedMeasurements = groupMeasurements(filteredMeasurements, groupBy, activeFilters[groupBy], groupingOrderedValues);
+
+  // Memoize D3 scale functions to allow deep comparison to work below for svgData
+  const xScale = useMemo(() => createXScale(width, measurements), [width, measurements]);
+  const yScale = useMemo(() => createYScale(), []);
+  // Memoize all data needed for basic SVG to avoid extra re-drawings
+  const svgData = useDeepCompareMemo({ xScale, yScale, x_axis_label, threshold, groupingOrderedValues, groupedMeasurements});
+  // Memoize handleHover function to avoid extra useEffect calls
+  const handleHover = useMemo(() => (data, dataType, mouseX, mouseY) => {
+    let newHoverData = null;
+    if (data !== null) {
+      // Create a Map of data to save order of fields
+      const newData = new Map();
+      if (dataType === "measurement") {
+        // Handle single measurement data
+        // Filter out internal auspice fields (i.e. measurementsJitter and measurementsId)
+        const displayFields = Object.keys(data).filter((field) => fields.has(field));
+        // Order fields for display
+        const fieldOrder = [...fields.keys()];
+        const orderedFields = orderBy(displayFields, (field) => fieldOrder.indexOf(field));
+        orderedFields.forEach((field) => {
+          newData.set(fields.get(field).title, data[field]);
+        });
+      } else if (dataType === "mean") {
```

Comment:
reminder to add a title to the on-hover panel (essentially the colour-by value, I think). We probably want to add the row-name (reference / group-by name) I think?

Reply:
Added as todo in #1463.